MSAL/src/MSALPublicClientApplication.m

```diff
@@ -890,6 +896,19 @@ - (void)updateExternalAccountsWithResult:(MSALResult *)result context:(id<MSIDRe
     }
 }
 
+- (BOOL)isAADNonConsumerTenant:(MSALAuthority *)authority
+{
+    if (![authority isKindOfClass:[MSALAADAuthority class]]) return NO;
+    
+    MSIDAuthority *msidAuthority = ((MSALAADAuthority *)authority).msidAuthority;
+    if (![msidAuthority isKindOfClass:[MSIDAADAuthority class]]) return NO;
+    
+    MSIDAADAuthority *aadMsidAuthority = (MSIDAADAuthority *)msidAuthority;
```

Comment:
nit: please consider not using force converting

Reply:
was checking at line 904, I have update the code with comments and structure for better readability.